airflow/models/taskinstance.py

```diff
@@ -94,6 +94,17 @@ def clear_task_instances(tis,
                 # outdated. We make max_tries the maximum value of its
                 # original max_tries or the current task try number.
                 ti.max_tries = max(ti.max_tries, ti.try_number - 1)
+
+            if ti.state == State.UP_FOR_RESCHEDULE:
```

Comment:
I am not sure about this line. Shouldn't it be clearable unrelated of its state? I put this `if` to only consider task instances that can have reschedules.

Reply:
Yeah I think we can remove the `if`

Reply:
Do you think I can even move it up out of the `if ti.state == State.RUNNING` right before it?